### DISCUSSION TITLE
How to use yabai effectively with many windows in a single space

### DISCUSSION THREAD

**Discussion by himat** [asmvik/yabai#1038] [Q&A] [answered]
I'm very new to yabai and this is my first tiling window manager. I just installed it a few minutes ago.

The thing is though that I normally have multiple chrome windows open on a single space, and so when I just installed yabai in bsp mode, it auto tiled every single window at half size intervals.
And so because I have so many windows, there are a bunch of windows that are now too small.
![image](https://user-images.githubusercontent.com/1156974/139520267-57b455ef-a057-47e8-960c-3f4aad435dbe.png)

As you can see, it's pretty useless to have so many tiny windows.

But the thing is that I probably only need 2 of these windows open at any time, and the others can be in the background until I switch to them with cmd/tilde+tab.

So what is the best way to use yabai here? Am I supposed to move most of my chrome windows into another desktop space? Or can I more ideally have them still in this space but in the background somehow?

**Comment by 24unix** [comment]
Why don’t you just minimise (command M) the unneeded ones?

**Comment by donaldguy** [comment, marked as answer]
It sounds like what you want is maybe `yabai -m config layout stack`, possibly with https://github.com/AdamWagner/stackline set up to easier keep track of what's where


but also you might just want to:
-  like @24unix said, minimize things. or hide things (Cmd-H) - hiding effects the whole app (so like ... all Chrome windows) but is faster. Also there is Alt-Cmd-H -> "Hide all others (everything except the focused app).
    
    If you aren't intent on using only Chrome, you could (with some overhead) run other browsers (most of which could also basically be[^1] chrome, e.g. [Chrome Canary](https://www.google.com/chrome/canary/), [Brave](https://brave.com), [Edge](https://www.microsoft.com/en-us/edge)) 
- use more spaces. If there are pairs of consistent useful windows that go together ... that's pretty definitely what you want
- Use Chrome tab groups (See: https://blog.google/products/chrome/manage-tabs-with-google-chrome/ ) or similar (e.g. Safari 15's tab groups) to achieve similar workflow with fewer windows. 
- You might also consider using fairly equivalently any of e.g. 
   - the <kbd>Shift</kbd>+<kbd>T</kbd> command from [Vimium](https://chrome.google.com/webstore/detail/vimium/dbepggeogbaibhgnhhndojpepiihcmeb?hl=en), 
   - the built-in `%` search in Firefox, 
   - Tab Finder (the Chrome [extension](https://chrome.google.com/webstore/detail/tab-finder/akikgmngikhdhecndjkjcacfefdbnnin?hl=en) or setapp standalone [app](https://setapp.com/apps/tabfinder)), 
   - An Alfred workflow like https://github.com/epilande/alfred-browser-tabs
   - A full-fledged keyboard-driven broswer like https://qutebrowser.org/ or https://nyxt.atlas.engineer/
   - etc.

    to make it so you can just ignore what window an open tab lives in and just jump to it based on the name/url (and thus get away with fewer windows the tabs are in without getting lost, then either ignore the tab bar or hide it using various hacks you can google for (you might also be interested in https://vivaldi.com which is a pretty vanilla (but [semi-](https://help.vivaldi.com/desktop/privacy/is-vivaldi-open-source/)closed source) Chrome fork that lets you hide tab bar natively or have a multi-row tab bar, do splits inside a single browser window, etc.)

[^1]: You can also probably _literally_ run multiple instances of Chrome which would hide individually using e.g. [`open -n`](https://ss64.com/osx/open.html) but that's likely to cause issues (unless you force them to use different Chrome profiles?) and be confusing (since they'll all probs have identical icons in Dock, Cmd-Tab, etc)

**Reply by himat** [reply]
Super comprehensive, thank you!! You're awesome.
